Paccmann vae inference (#175)
GT4SD/gt4sd-core: examples/molecular_vae/README.md

# Molecular VAE

### A short tutorial on training a molecular VAE to generate SMILES and subsequently use it for inference

#### Training 
This trains a molecular VAE using encoder and decoder based on StackGRUs. For details, see the PaccMannRL papers, for [omics](https://www.sciencedirect.com/science/article/pii/S2589004221002376) and [proteins](https://iopscience.iop.org/article/10.1088/2632-2153/abe808/meta).

> ```

Reviewer: ```suggestion
```console
```

conda activate gt4sd
gt4sd-trainer  --training_pipeline_name paccmann-vae-trainer --epochs 250 --batch_size 4 --n_layers 1 --rnn_cell_size 16 --latent_dim 16 --train_smiles_filepath src/gt4sd/training_pipelines/tests/molecules.smi --test_smiles_filepath src/gt4sd/training_pipelines/tests/molecules.smi --model_path /tmp/gt4sd-paccmann-gp/ --training_name fast-example --eval_interval 15 --save_interval 15 --selfies

> ```

Reviewer: ```suggestion
```

```

> *NOTE*: You might want to pass a SMILES/SELFIES language object via `--smiles_language_filepath`

Reviewer: here I'm afraid the suggestion did not go through as expected, probably the three quotes are not being escaped

Author: fixed now

*NOTE*: This is *dummy* example, do not use "as is" :warning:

> #### Inference

Reviewer: ```suggestion
#### Inference

```

This training pipeline trains a molecular VAE. The GT4SD **inference** pipeline support inference for PaccMannRL, i.e., a hybrid-VAE with an omics or a protein encoder and a molecular decoder.

> There's not (yet) a dedicated inference pipeline for a normal molecular VAE, hence `gt4sd-saving` will not work (you could train the GuacaMol VAE instead via GT4SD).

Reviewer: ```suggestion
There's not (yet) a dedicated inference pipeline for a plain molecular VAE (you could train the GuacaMol VAE via GT4SD or use the controlled sampling approach in the PaccMannGP inference pipeline).
```

> There's not (yet) a dedicated inference pipeline for a plain molecular VAE (you could train the GuacaMol VAE via GT4SD or use the controlled sampling approach in the PaccMannGP inference pipeline).

Reviewer: ```suggestion
There's not (yet) a dedicated inference pipeline for a plain molecular VAE (you could train the GuacaMol VAE via GT4SD or use `gt4sd-saving` for the controlled sampling approach from PaccMannGP inference pipeline).
```

To use a molecular VAE trained with `gt4sd-trainer  --training_pipeline_name paccmann-vae-trainer`, do the following:

``` sh
cd ~/.gt4sd/algorithms/conditional_generation/PaccMannRL/PaccMannRLProteinBasedGenerator/

> cp -r v0 your_model_name # if this path does not exist, trigger the download via running the example from the README

Reviewer: ```suggestion
cp -r v0 ${YOUR_MODEL_NAME} # if this path does not exist, trigger the download via running the example from the README
```

> cd your_model_name

Reviewer: ```suggestion
cd ${YOUR_MODEL_NAME}
```

# overwrite the weights file (.pt) with your desired checkpoint

> cp path_to_your_model_file selfies_conditional_generator.pt  # use that filename even if you dont do SELFIES

Reviewer: ```suggestion
cp ${PATH_TO_YOUR_MODEL_FILE} selfies_conditional_generator.pt  # use that filename even if you don't do SELFIES
```

# overwrite the selfies_language with the one from your model

> cp path_to_your_smiles_language selfies_language.pkl

Reviewer: ```suggestion
cp ${PATH_TO_YOUR_SMILES_LANGUAGE} selfies_language.pkl
```

# Use PaccMann molecular VAE from a PaccMannRL inference pipeline

### A short tutorial on training a molecular VAE to generate SMILES and subsequently use it for inference

#### Training

This trains a molecular VAE using encoder and decoder based on StackGRUs. For details, see the PaccMannRL papers, for [omics](https://www.sciencedirect.com/science/article/pii/S2589004221002376) and [proteins](https://iopscience.iop.org/article/10.1088/2632-2153/abe808/meta).

```console
conda activate gt4sd
gt4sd-trainer  --training_pipeline_name paccmann-vae-trainer --epochs 250 --batch_size 4 --n_layers 1 --rnn_cell_size 16 --latent_dim 16 --train_smiles_filepath src/gt4sd/training_pipelines/tests/molecules.smi --test_smiles_filepath src/gt4sd/training_pipelines/tests/molecules.smi --model_path /tmp/gt4sd-paccmann-gp/ --training_name fast-example --eval_interval 15 --save_interval 15 --selfies
```

> *NOTE*: You might want to pass a SMILES/SELFIES language object via `--smiles_language_filepath`

Reviewer: ```suggestion

*NOTE*: You might want to pass a SMILES/SELFIES language object via `--smiles_language_filepath`
```
